src/ig/IGExporter.ts

```diff
@@ -444,6 +456,54 @@ export class IGExporter {
     }
   }
 
+  /**
+   * Adds generated pages for each artifact type included in the IG.
+   * Only add pages if there is not already an existing page with the name that would be added.
+   */
+  private addArtifactPages(igPath: string): void {
+    // profiles page has a table with 3 columns:
+    // Profile, Derived From, Description
+    // other pages have tables with 2 columns:
+    // [Type], Description
+    const pageContentExportPath = path.join(igPath, 'input', 'pagecontent');
+    ensureDirSync(pageContentExportPath);
+    for (const artifactType of Object.keys(this.artifacts) as (keyof ArtifactCollection)[]) {
+      const artifactPagePath = path.join(pageContentExportPath, `${artifactType}.md`);
+      if (existsSync(artifactPagePath)) {
+        // overwriting the file is okay only if the existing file looks like it was generated by this function
+        const existingContent = readFileSync(artifactPagePath, 'utf-8');
+        if (!existingContent.startsWith('<!-- Generated by SUSHI -->')) {
+          logger.error(
+            `A page named ${artifactType}.md already exists. No artifact page will be created at this path.`
+          );
+          continue;
+        }
+      }
+      let content = '<!-- Generated by SUSHI -->\n';
```

Comment:
When I run the IG publisher with the generated file, it looks like the markdown isn't being properly rendered when I just have the comment and the table in the file. If I add a blank line between the comment and the table, it renders fine. It also is fine if there is a line of text before the table as well, which could mirror what mCODE creates by hand if we added a line saying "The following ___ have been defined for this implementation guide" (see mCODE example [here](https://build.fhir.org/ig/HL7/fhir-mCODE-ig/profiles.html)). I think either works, I have a slight preference for the text just to give some context to the page, but I have no strong feeling either way.